app/validation.py

```diff
@@ -60,11 +61,12 @@ def get_validator(schema_name, enforce_required=True, required_fields=None):
     if enforce_required:
         schema = _SCHEMAS[schema_name]
     else:
-        schema = _SCHEMAS[schema_name].copy()
+        schema = copy.deepcopy(_SCHEMAS[schema_name])
```

Comment:
`deepcopy` is slower since it has to do quite a bit more work, and the intent is not as clear when we're only modifying the top-level. But other than that it should work, so we can keep it if you think we're likely to need it in the future.